modules/local/reveal/filter_quality.nf

```diff
@@ -12,7 +12,7 @@ process FILTER_QUALITY {
     tuple val(meta), path(tab) // sequence tsv in AIRR format
 
     output:
-    tuple val(meta), path("*quality-pass.tsv"), emit: tab // sequence tsv in AIRR format
+    tuple val(meta), path("*quality-pass.tsv"), optional:true, emit: tab // sequence tsv in AIRR format
```

Comment:
```suggestion
    tuple val(meta), path("*quality-pass.tsv"), emit: tab // sequence tsv in AIRR format
```

I would not make this file optional, I would let it fail if the output table is not produce, so that the user realizes that there are no sequences left after filtering for this process in particular.

Reply:
In the final size report, where we show the number of reads at each step, the user should be able to see where the sample disappeared. My rationale making this output optional was that there is going to be cases where the file is not produced, because 0 sequences pass the quality filters, but this is not an error, it just is the end of the road for that sample. This case is unlikely to happen with airr-seq repertoires, but I bumped into this in a project where I am using rna-seq+trust4 repertoires. For some samples I had just 2 or 3 sequences, and they didn't pass the quality filters. FILTER_QUALITY created an empty quality-pass.tsv file and then CHANGEO_PARSEDB_SPLIT failed because ParseDb.py can't process an empty file, and then the workflow failed. I added errorStrategy = 'ignore' in my .config file for these processes, but that seemed to me that would ignore true errors and I thought using optional was more "truthful" and wouldn't require adding things in .conf. I don't know what the best option is. I just didn't like to get an error when there was no error. And maybe in other filtering steps a similar thing could happen.

Reply:
That's true and a good reason, I am hesitant usually to add `optional: True` to the main output of a process as it is then easy to miss errors in case a tool silently fails, but I guess then it is rather the problem of the tool not properly throwing errors. Ok let's merge this as is for now and we can think about other filtering processes that would require this `optional: True` statement as well.